lightning/src/ln/channel.rs

```diff
@@ -1026,10 +1079,24 @@ pub(super) struct ChannelContext<SP: Deref> where SP::Target: SignerProvider {
 	/// This flag is set in such a case. Note that we don't need to persist this as we'll end up
 	/// setting it again as a side-effect of [`Channel::channel_reestablish`].
 	signer_pending_commitment_update: bool,
+	/// Similar to [`Self::signer_pending_commitment_update`]: indicates that we've deferred sending a
+	/// `revoke_and_ack`, and should do so once the signer has become unblocked.
+	signer_pending_revoke_and_ack: bool,
 	/// Similar to [`Self::signer_pending_commitment_update`] but we're waiting to send either a
 	/// [`msgs::FundingCreated`] or [`msgs::FundingSigned`] depending on if this channel is
 	/// outbound or inbound.
 	signer_pending_funding: bool,
+	/// Similar to [`Self::signer_pending_commitment_update`] but we're waiting to send a
+	/// [`msgs::ChannelReady`].
+	signer_pending_channel_ready: bool,
+	/// If we attempted to retrieve the per-commitment point for the next transaction but the signer
+	/// wasn't ready, then this will be set to `true`.
+	signer_pending_commitment_point: bool,
+	/// If we attempted to release the per-commitment secret for the previous transaction but the
+	/// signer wasn't ready, then this will be set to `true`.
+	signer_pending_released_secret: bool,
+	/// If we need to send a `channel_reestablish` then this will be set to `true`.
+	signer_pending_channel_reestablish: bool,
```

Comment:
I feel like there was a reason that we wanted to be able to get the next per-commitment point without releasing the previous commitment secret, but I'm not remembering the details.

Agreed that this has gotten messy. Let me see if I can drive it to correctness and then maybe we can work through what it might take to refactor and simplify.